classes/helper/login_helper.php

```diff
@@ -531,4 +532,19 @@ public function clear_username_in_auth_global_vars() {
         $frm = null;
         set_moodle_cookie('');
     }
+
+    /**
+     * Get user failed login count
+     *
+     * @param $userName
+     * @return int|null
+     * @throws \coding_exception
+     * @throws \dml_exception
+     */
+    public function get_user_login_failCount($userName)
+    {
+        global $DB;
+        $userTemp = $DB->get_record('user', ['username' => $userName]);
```

Comment:
How much extra work do we think it would be to store the users id in the session cache so we dont query for the same thing if the user keeps trying? 

Also looking at the get_user_preferences() function it will get the $USER for you if you dont pass one. Not sure if one would be more efficient than another?

Reply:
Or, given this plugin already uses globals you could just use $USER.